dd-sdk-android/src/test/kotlin/com/datadog/android/sessionreplay/internal/SessionReplayFeatureTest.kt

```diff
@@ -53,4 +64,145 @@ internal class SessionReplayFeatureTest : SdkFeatureTest<Any, Configuration.Feat
         assertThat(testedFeature.persistenceStrategy)
             .isInstanceOf(SessionReplayRecordPersistenceStrategy::class.java)
     }
+
+    @Test
+    fun `M register the Session Replay lifecycle callback W initialize()`() {
+        // When
+        testedFeature.initialize(appContext.mockInstance, fakeConfigurationFeature)
+
+        // Then
+        verify(mockSessionReplayLifecycleCallback)
+            .register(appContext.mockInstance)
+    }
+
+    @Test
+    fun `M unregister the Session Replay lifecycle callback W stop()`() {
+        // Given
+        testedFeature.initialize(appContext.mockInstance, fakeConfigurationFeature)
+
+        // When
+        testedFeature.stop()
+
+        // Then
+        verify(mockSessionReplayLifecycleCallback)
+            .unregisterAndStopRecorders(appContext.mockInstance)
+    }
+
+    @Test
+    fun `M unregister the SessionReplayCallback W stopRecording() { was recording }`() {
+        // Given
+        testedFeature.initialize(appContext.mockInstance, fakeConfigurationFeature)
+
+        // When
+        testedFeature.stopRecording()
+
+        // Then
+        verify(mockSessionReplayLifecycleCallback)
+            .register(appContext.mockInstance)
+    }
+
+    @Test
+    fun `M unregister only once the SessionReplayCallback W stopRecording() { multi threads }`() {
+        // Given
+        val countDownLatch = CountDownLatch(3)
+        testedFeature.initialize(appContext.mockInstance, fakeConfigurationFeature)
+
+        // When
+        repeat(3) {
+            Thread {
+                testedFeature.stopRecording()
+                countDownLatch.countDown()
+            }.run()
+        }
+
+        // Then
+        countDownLatch.await(5, TimeUnit.SECONDS)
+        verify(mockSessionReplayLifecycleCallback)
+            .register(appContext.mockInstance)
```

Comment:
seems like verification of `unregister` is missing

Reply:
yup, copy pasted from the first one :)